### DISCUSSION TITLE
SSH node service - standard mode VS reverse tunnel mode

### DISCUSSION THREAD

**Discussion by alwaysastudent** [gravitational/teleport#19138] [Q&A] [answered]
Hi folks, I know we can configure our 'node service' in two ways: standard mode or reverse tunnel mode. My question is, which setup is more secure? Or do they both have the same security posture? Also, performance wise which is better?

**Comment by travelton** [comment, marked as answer]
There are security considerations to be aware of, particularly when joining a new node to a Teleport Cluster. Less so once the node has established its connection with the Teleport Cluster. 

Given `standard mode` is generally deployed in environments where the network connections are on a trusted private network, you can be reasonably sure the Teleport Cluster that you're connecting to is the expected Teleport Cluster. When using `reverse tunnel mode`, the node may need to traverse an untrusted public network. 

However, in both cases, Teleport recommends utilizing the functionality to [pin a CA](https://goteleport.com/docs/management/admin/adding-nodes/#obtain-a-ca-pin). This protects against attempts to MITM the initial connection when [joining nodes](https://goteleport.com/docs/architecture/nodes/#joining-nodes) to your Teleport Cluster. 

Once a node is joined to the Teleport Cluster, the underlying established network connection, regardless of connection method, offers the same level of transport security. Under the hood, an SSH tunnel is established. This tunnel is fully encrypted. 

One last note on security related to node joining... Be sure that you generate low-TTL node join tokens. Using pre-defined static tokens in your Teleport Auth configuration is not recommended for production environments. Review the `tctl tokens add` documentation [here](https://goteleport.com/docs/reference/cli/#tctl-tokens-add). 

Performance-wise, you likely won't notice much difference until you reach ~10k nodes. Please see our [scaling documentation](https://goteleport.com/docs/management/operations/scaling/) for more performance information. Though, be aware that traversing public networks may introduce uncontrollable latency due to network congestion or additional hops to reach the Teleport Cluster.

**Reply by alwaysastudent** [reply]
@travelton thank you for the reply. It gives a better understanding. I have a follow-up question though,

Would `reverse tunnel mode` eliminate all direct calls to the auth service from the node (If so then, would it give any added benefits since we don't have to expose the auth service to anyone else but the proxy)? 

How does the ssh session recording happen with the reverse tunnel mode?

**Reply by travelton** [reply]
@alwaysastudent No problem! 

> Would reverse tunnel mode eliminate all direct calls to the auth service from the node (If so then, would it give any added benefits since we don't have to expose the auth service to anyone else but the proxy)?

The node will communicate with the auth server through the proxy, eliminating direct connections to the auth server itself.

> How does the ssh session recording happen with the reverse tunnel mode?

By default, session recording occurs on the node itself. Once the session concludes, the node will send the recording through the proxy to the auth server for storage. You can change this by enabling [Proxy Recording Mode](https://goteleport.com/docs/server-access/guides/recording-proxy-mode/).
